patches/minecraft/net/minecraft/world/gen/layer/BiomeLayer.java.patch

```diff
@@ -0,0 +1,75 @@
+       }
+ 
++      int desertIdx = net.minecraftforge.common.BiomeManager.BiomeType.DESERT.ordinal();
++
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76769_d, 30));
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_150588_X, 20));
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76772_c, 10));
++
++      if (p_i232147_1_) {
++         biomes[desertIdx].clear();
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76769_d, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76767_f, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76770_e, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76780_h, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76772_c, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76768_g, 10));
++      }
+    }
+ 
+    public int func_202726_a(INoiseRandom p_202726_1_, int p_202726_2_) {
+@@ -46,21 +66,21 @@
+                return p_202726_1_.func_202696_a(3) == 0 ? field_202733_g : field_202734_h;
+             }
+ 
+-            return this.field_151623_c[p_202726_1_.func_202696_a(this.field_151623_c.length)];
++            return Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.DESERT, p_202726_1_).biome);
+          case 2:
+             if (i > 0) {
+                return field_202732_f;
+             }
+ 
+-            return field_202745_s[p_202726_1_.func_202696_a(field_202745_s.length)];
++            return Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.WARM, p_202726_1_).biome);
+          case 3:
+             if (i > 0) {
+                return field_202737_k;
+             }
+ 
+-            return field_202746_t[p_202726_1_.func_202696_a(field_202746_t.length)];
++            return Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.COOL, p_202726_1_).biome);
+          case 4:
+-            return field_202747_u[p_202726_1_.func_202696_a(field_202747_u.length)];
++            return Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.ICY, p_202726_1_).biome);
+          default:
+             return field_202735_i;
+          }
+@@ -68,4 +88,11 @@
+          return p_202726_2_;
+       }
+    }
++
++   protected net.minecraftforge.common.BiomeManager.BiomeEntry getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType type, INoiseRandom context) {
```

Comment:
You need to add the Forge method markers. Additionally please check that this is at the end of the file.

Reply:
This is at the end of the file, what are Forge method markers @OrionDevelopment ?

Reply:
Look at other patched vanilla classes which have methods that have been added by forge.
There are two comments: ///////// FORGE START /////////// and /////////// FORGE END //////////// or something in that direction.
These enclose the methods that forge added to that class, and allow us to easily identify the methods which are not from the vanilla code base and are very handy during porting.

Reply:
Ah, makes sense.  Done.